crates/ide_db/src/helpers/import_assets.rs

```diff
@@ -25,16 +30,22 @@ pub enum ImportCandidate {
 
 #[derive(Debug)]
 pub struct TraitImportCandidate {
-    pub receiver_ty: hir::Type,
+    pub receiver_ty: Type,
     pub name: NameToImport,
 }
 
 #[derive(Debug)]
 pub struct PathImportCandidate {
-    pub qualifier: Option<ast::Path>,
+    pub qualifier: Qualifier,
     pub name: NameToImport,
 }
 
+#[derive(Debug)]
+pub enum Qualifier {
+    Absent,
+    FirstSegmentUnresolved(ast::NameRef, ModPath),
```

Comment:
Let's maybe make this a record rather that a tuple? Recently, I've noticed that everything that starts as a tuple almost invariably gets refactord to the record down the road and, even before that, records just seem easier to work with? Right now, I'd love to search for all the usages of the second field here, and, with a tuple, there isn't a field I can just click on.

Reply:
I've replaced `Absent` with `Option` usage and got it slightly simpler in the end.

Reply:
Yeah, it's also true that using option is most of the time preferable to introducing an explicit None variant.